client/src/hooks/useApplicationData.jsx

```diff
@@ -0,0 +1,71 @@
+import React, { useEffect, useReducer } from "react";
+import reducer from "../reducers/application";
+
+import shuffleArray from "../helpers/shuffleArray";
+
+const baseURL = process.env.REACT_APP_BASE_URL;
+
+const useApplicationData = () => {
+    const [state, dispatch] = useReducer(reducer, {
+        carouselSelectedIndex: 0,
+        carouselImages: [],
+        category: "both"
+    });
+
+    useEffect (() => {
+        fetchImageCategory("cats")
+    }, []);
+
+    const changeCarouselSelection = indexChange => dispatch({type: "changeSelection", indexChange});
+
+    const changeCategory = category => {
+        dispatch({type: "changeCategory", category})
+        if (category === "both") {
+            return fetchBothCategories();
+        } 
+        return fetchImageCategory(category);
+    };
+
+    const fetchImageCategory = category => {
+        return new Promise((resolve, reject) => {
+            return fetch(`${baseURL ? baseURL : ""}/api/${category}`)
+            .then(res => res.json())
+            .then(async images => {
+                await dispatch({
+                    type: "updateCarousel",
+                    images
+                });
+                resolve(images)
+            })
+            .catch(err => {
+                reject(err)
+            });
```

Comment:
Any reason you chose to chain promises over async/await?

Reply:
There was no particular reason that I preferred the use of promise chaining in this instance. I definitely could see that using async await could have been more readable and could have achieved the same effect by containing within a sincle try catch block. This decision was just a choice out of personal preference at the time but would definitely be something I could change.